### DISCUSSION TITLE
RAG failing to extract information

### DISCUSSION THREAD

**Discussion by gelim** [danny-avila/LibreChat#3520] [Troubleshooting] [answered]
### What happened?

Hello,

I'm testing the rag-api with text-embedding-3-small hosted on Azure.
It's kinda working, but when I want to extract some specific information it will give me only a subset 
(if not failing overall)
Is there a window limit that could be tweaked somehow?

Tests done with Llama3.1 8B & 70B.

### Steps to Reproduce

Example document: [OSTIF-QuarksLab-Monero-Bulletproofs-Final2.pdf](https://github.com/juliocesarfort/public-pentesting-reports/blob/master/QuarksLab/OSTIF-QuarksLab-Monero-Bulletproofs-Final2.pdf)

Example query:

"extract all the critical severity vulnerabilities": all properly extracted
"extract all the medium severity vulnerabilities" : none found

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This is what RAG does, it only finds a subset of information. It's good for needle-in-a-haystack type search, not an exhaustive search.

That being said, there can definitely be improvements and enhancements to increase general performance.

**Reply by danny-avila** [reply, marked as answer]
More info on how you can tweak the current behavior: https://www.librechat.ai/docs/configuration/rag_api
